### ISSUE TITLE
Question: Is it possible to have one test and run it on all variables from one folder?

### ISSUE THREAD

**Issue by Starday2009** [promptfoo/promptfoo#1793]
Question: Is it possible to have one test and run it on all variables from one folder?

I’m trying to run it this way — I have one file with the test and several files with different data, but when I run it like this, only one test gets executed:

```
- vars:
    ../../vars/*
  assert:
    - type: python
      value: file://scripts/test_data_in_output.py
      metric: deterministic
```

**Comment by typpo** [comment]
Yes, the summarization example has an example of this: https://github.com/promptfoo/promptfoo/blob/main/examples/summarization/promptfooconfig.yaml

**Comment by Starday2009** [comment]
Hi! Thank you! 
But when I changed the configuration to this

tests:
  - vars:
      data: file://data/*
    assert:
      - type: python
        value: file://scripts/test_data_in_output.py
        metric: deterministic

My tests stopped seeing the contents of the file with the variables. The file has the following structure and has .yaml type

DOC_ID: "995"
ENVIRONMENT: QA
report_date: "2022"

I retrieve the variables in this way:

context['vars']['ENVIRONMENT']

**Comment by mldangelo** [comment]
Hi @Starday2009, unfortunately, the variables from your YAML file are being loaded as a single variable, which is causing the issue you're seeing. To resolve this, you can use an `inputTransform` to map the variables into an object before running the tests. More details can be found in our documentation: https://www.promptfoo.dev/docs/configuration/guide/#transforming-input-variables (thanks @MrFlounder).

Alternatively, you can handle it in a custom prompt function by using `JSON.parse` or `json.loads` on `context.vars.data`: https://www.promptfoo.dev/docs/configuration/parameters/#prompt-functions.

Happy to hop on a call if you need help implementing this!

**Comment by Starday2009** [comment]
Hi @mldangelo! The first option seemed ideal to me. However, for some reason, the transform_vars.py does not recognize the contents of vars. 
Here is what I’m trying to do

```
def get_transform_vars(vars, context):
    return {
        'file_content': vars
    }
```

with 

```
tests:
  - options:
      transformVars: file://scripts/transform_vars.py:get_transform_vars
  - vars:
      file_data: "file://vars/*.yaml"
```

 and what I’m getting:
<img width="953" alt="Screenshot 2024-10-06 at 20 27 07" src="https://github.com/user-attachments/assets/9b17baca-0bbc-47d2-9345-8d542f11024a">

Maybe I'm doing something wrong?
When I tried to do it by reading the file, like this:

```
def get_transform_vars(vars, context):
    folder_path = os.getcwd() 
    file_path = os.path.join(folder_path, vars['file_path']) 

    with open(file_path, 'r') as file:
        file_content = file.read()

    return {
        'file_content': file_content,
    }
```

 it also didn't work, since the full path to the file is expected, and not /*
```

  - vars:
      file_data: "file://vars/*.yaml"
```

**Comment by mldangelo-oai** [comment]
Thanks for the question and for the detailed follow-up examples. I’m closing this thread as answered/superseded by the current supported data-loading patterns.

For the original use case — running the same test logic across many inputs from files — Promptfoo supports expanding tests from external test files and glob patterns, for example via `tests: file://.../*.yaml`. For cases where file-backed content needs to be reshaped before evaluation, the documented transform path is the intended escape hatch.

The later nested-`vars` glob behavior discussed in the thread is a slightly different concern from the original question. If that still behaves unexpectedly on current Promptfoo, a fresh minimal repro as a separate bug issue would be the best way to track it cleanly.
